### DISCUSSION TITLE
All unique values in columns

### DISCUSSION THREAD

**Discussion by zjael** [pocketbase/pocketbase#2984] [Q&A]
How can I get all the unique values from multiple columns in a collection let's call it "Products", with the columns "Brand, model, category", with the count for each if possible.

The purpose is for handling filters on the frontend, with a dropdown for each of the above columns, and the possible values.

**Comment by ganigeorgiev** [comment]
There is no builtin mechanism for this. Instead you can try to create a helper [View collection](https://pocketbase.io/docs/collections/) with the aggregation that you need.

I'm not sure that I understand the requirements but your view query could look something like:

```sql
SELECT
  (ROW_NUMBER() OVER()) as id,
  brand,
  model,
  category,
  count("*") as total
FROM products
GROUP BY (brand, model, category)
```

Keep in mind that views can't be parameterized, aka. if you want to be able to change the above 3 group columns dynamically, you'll have to register a [custom route](https://pocketbase.io/docs/router/#register-new-route) with the desired query (_with the upcoming v0.17 there will be more detailed guides and also JS hooks support_).

**Reply by ganigeorgiev** [reply]
_On second read, if you want the distinct items count for **each** field, not the combined unique of the 3 fields, then the query could be something like this (I haven't tested it and I'm not sure how well it will perform):_

```sql
SELECT
  (ROW_NUMBER() OVER()) as id,
  count(distinct brand) totalBrands,
  count(distinct model) totalModels,
  count(distinct category) totalCategories
FROM products
```
